io-engine/src/subsys/nvmf/subsystem.rs

```diff
@@ -165,6 +232,132 @@ fn make_sn<T: AsRef<[u8]>>(uuid: T) -> CString {
+ 
+            info!("controller successfully reset in response to I/O timeout for nexus ");
+        } else {
+            error!("failed to reset controller in response to I/O timeout");
+        }
+    }
+
+    extern "C" fn nvmf_event_handler(
+        subsystem: *mut spdk_nvmf_subsystem,
+        cb_arg: *mut c_void,
+        event: spdk_nvmf_subsystem_events,
+    ) {
+        let ss = NvmfSubsystem::from(subsystem);
+        let subsys_nqn = unsafe {
+            spdk_nvmf_subsystem_get_nqn(ss.0.as_ptr())
+                .as_str()
+                .to_string()
+        };
+        debug!(
+            "nvmf_event_handler called {:?} for subsys {:?}",
+            event, subsys_nqn
+        );
+        // let spdk_ctrlr_ptr = cb_arg as *mut spdk_nvmf_ctrlr;
+        let spdk_ctrlr = SpdkNvmfController::from(cb_arg as *mut spdk_nvmf_ctrlr);
+
+        debug!("host controler nqn {:?} ", spdk_ctrlr);
+
+        let nexus_name = match NvmfSubsystem::extract_nexus_name(&subsys_nqn) {
+            Some(value) => value,
+            None => {
+                debug!("subsys {:?} is not a valid nexus", subsys_nqn);
+                return;
+            }
+        };
+
+        let nexus = match nexus_lookup_mut(&nexus_name) {
+            Some(v) => v,
+            None => {
+                debug!("subsys {:?} is not having valid nexus", subsys_nqn);
+                return;
+            }
+        };
+        debug!("Nexus found for subsys {:?}", subsys_nqn);
+
+        match event {
+            SPDK_NVMF_SS_INIATOR_TIMEOUT => {}
+            // SPDK_NVMF_SS_INIATOR_CONNECT |  // TODO Need to be handled
+            // SPDK_NVMF_SS_INIATOR_DISCONNECT | //TODO
+            _ => {
+                debug!("event {:?} is not handled ", event);
+                return;
+            }
+        }
+        // let rt = tokio::runtime::Builder::new_current_thread()
+        // .enable_all()
+        // .build()
+        // .unwrap();
+        // let res = tokio::spawn( async move { ss.pause() });                     
+
+        for (idx, child) in nexus.children_iter().enumerate() {
+            debug!(
+                "Nexus found a {:?} at index {:?} for subsys {:?}",
+                child, idx, subsys_nqn
+            );
+            let dev = match child.get_device() {
+                Ok(dev) => dev,
+                Err(error) => {
+                    debug!(
+                        "subsys {:?} get_device failed for {:?} with Error {:?}",
+                        subsys_nqn, child, error
+                    );
+                    return;
+                }
+            };
+            let dev_name = dev.device_name();
+            debug!(
+                "nvmf_event_handler: device name {:?} - nexus subsys {:?}",
+                dev_name, subsys_nqn
+            );
+            let ctrlr = match NVME_CONTROLLERS.lookup_by_name(&dev_name) {
+                Some(ctrlr) => ctrlr,
+                None => {
```

Comment:
Such a situation can take place when nexus has local child bdev replicas. We should skip such devices.